inc/blocks/namespace.php

```diff
@@ -72,11 +73,12 @@ function on_save_post( int $post_ID, WP_Post $post, bool $update ) : void {
 		return;
 	}
 
-	if ( $post->post_type === POST_TYPE ) {
+	if ( in_array( $post->post_type, [ POST_TYPE, 'revision', 'customize_changeset' ], true ) ) {
 		return;
 	}
 
-	if ( $post->post_type === 'revision' ) {
+	$post_type_object = get_post_type_object( $post->post_type );
+	if ( ! $post_type_object->public || ! $post_type_object->show_in_rest ) {
```

Comment:
We only want to return early if both of these are false
```suggestion
	if ( ! $post_type_object->public && ! $post_type_object->show_in_rest ) {
```